### PR TITLE
Sebastian

### PR DESCRIPTION
**Pull request by scriptkitty**

### CODE REVIEW

File: test/zombiefactory.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	const account_two = accounts[1];
+	const name = "Name"
+	let contract;
+
+	beforeEach(async () => {
+        contract = await ZombieFactory.deployed();
+	});
+
+	it("extends the zombies array by exactly one with createRandomZombie", async () =>  {
+		let preLength = (await contract.getNumberOfZombies.call()).toNumber()
```
Comment (alexlatif) [inline test/zombiefactory.js]: The need to be const as it is never reassigned

File: test/zombiefactory.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		await contract.createRandomZombie(name, {from: account_one})	
+		let postLength = (await contract.getNumberOfZombies.call()).toNumber()
```
Comment (alexlatif) [inline test/zombiefactory.js]: here

File: test/zombiefactory.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		assert.equal(preLength + 1, postLength, "createRandomZombie did not increase the array length by one.");
+	});
+
+	it("creates a new zombie with the given name in createRandomZombie", async () => {
+		let zombie = (await contract.zombies.call(0))
```
Comment (alexlatif) [inline test/zombiefactory.js]: here

File: test/zombiefactory.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+		// The call does not return the struct but rather an array of members.
+		// Since "name" is the first member of the struct it is the first entry of the array.
+		assert.equal(name, zombie[0], "createRandomZombie did not create a zombie with the specified name.")
+
+	});
+
+	it("creates zombies with equal DNA for equivalent names", async () => {
+		await contract.createRandomZombie(name, {from: account_two})
+		let oldZombie = (await contract.zombies.call(0))
```
Comment (alexlatif) [inline test/zombiefactory.js]: here

File: test/zombiefactory.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		let newZombie = (await contract.zombies.call(1))
```
Comment (alexlatif) [inline test/zombiefactory.js]: here

File: test/zombiefactory.js (hunk 1 above)
Comment (alexlatif) [reply test/zombiefactory.js]: also what are the brackets around the await for?

**Review by alexlatif — CHANGES_REQUESTED** [review]
Need to fix conflict and clean commits

File: test/zombiefactory.js (hunk 1 above)
Comment (scriptkitty) [reply test/zombiefactory.js]: Brackets are for the .toNumber() call

**Review by alexlatif — APPROVED** [review]

**Pull request merged**